handler/oauth2/flow_authorize_code_auth_test.go

```diff
@@ -122,11 +122,40 @@ func TestAuthorizeCode_HandleAuthorizeEndpointRequest(t *testing.T) {
 					},
 					description: "should pass",
 					expect: func(t *testing.T, areq *fosite.AuthorizeRequest, aresp *fosite.AuthorizeResponse) {
-						code := aresp.GetQuery().Get("code")
+						code := aresp.GetParameters().Get("code")
 						assert.NotEmpty(t, code)
 
-						assert.Equal(t, strings.Join(areq.GrantedScope, " "), aresp.GetQuery().Get("scope"))
-						assert.Equal(t, areq.State, aresp.GetQuery().Get("state"))
+						assert.Equal(t, strings.Join(areq.GrantedScope, " "), aresp.GetParameters().Get("scope"))
+						assert.Equal(t, areq.State, aresp.GetParameters().Get("state"))
+					},
+				},
+				{
+					areq: &fosite.AuthorizeRequest{
+						ResponseTypes: fosite.Arguments{"code"},
+						Request: fosite.Request{
+							Client: &fosite.DefaultClient{
+								ResponseTypes: fosite.Arguments{"code"},
+								RedirectURIs:  []string{"https://asdf.de/cb"},
+								Audience:      []string{"https://www.ory.sh/api"},
+							},
+							RequestedAudience: []string{"https://www.ory.sh/api"},
+							GrantedScope:      fosite.Arguments{"a", "b"},
+							Session: &fosite.DefaultSession{
+								ExpiresAt: map[fosite.TokenType]time.Time{fosite.AccessToken: time.Now().UTC().Add(time.Hour)},
+							},
+							RequestedAt: time.Now().UTC(),
+						},
+						State:       "superstate",
+						RedirectURI: parseUrl("https://asdf.de/cb"),
+					},
+					description: "Default responseMode check",
```

Comment:
To me it looks like the previous tests all cover the "default responseMode" case. I think we should probably add the explicit query response mode and also the fragment response mode as a test!

Reply:
+1